src/Cake.Issues.Reporting.Sarif.Tests/SarifIssueReportGeneratorTests.cs

```diff
@@ -998,70 +979,70 @@ public void Should_Generate_Report_With_BaseLineState_Updated_If_BaselineGuid_Se
+
+            var issues =
                 new List<IIssue>
                 {
                     IssueBuilder
                         .NewIssue("Message Foo.", "ProviderType Foo", "ProviderName Foo")
-                        .InFile(@"src\Cake.Issues.Reporting.Sarif.Tests\SarifIssueReportGeneratorTests.cs", 23, 42, 5, 10)
+                        .InFile(@"src\Cake.Issues.Reporting.Sarif.Tests\SarifIssueReportGeneratorTests.cs", 123, 142, 15, 110)
                         .Create(),
+                };
 
+            var existingIssues =
+                new List<IIssue>
+                {
                     IssueBuilder
                         .NewIssue("Message Foo.", "ProviderType Foo", "ProviderName Foo")
                         .InFile(@"src\Cake.Issues.Reporting.Sarif.Tests\SarifIssueReportGeneratorTests.cs", 125, 144, 15, 110)
                         .Create(),
-
-                    IssueBuilder
-                        .NewIssue("Message Foo Bar.", "ProviderType Foo", "ProviderName Foo")
-                        .InFile(@"src\Cake.Issues.Reporting.Sarif.Tests\SarifIssueReportGeneratorTests.cs", 123, 142, 15, 110)
-                        .Create(),
                 };
 
-            fixture.SarifIssueReportFormatSettings.ExistingIssues.AddRange(existinIssues);
+            fixture.SarifIssueReportFormatSettings.ExistingIssues.AddRange(existingIssues);
 
             // When
             var logContents = fixture.CreateReport(issues);
 
             // Then
             var sarifLog = JsonConvert.DeserializeObject<SarifLog>(logContents);
 
-            // There are two runs because there are two issue providers.
-            sarifLog.Runs.Count.ShouldBe(1);
-
-            var run = sarifLog.Runs[0];
-            run.Tool.Driver.Name.ShouldBe("ProviderType Foo");
-            run.Results.Count.ShouldBe(3);
-
-            var result = run.Results[0];
-            result.Message.Text.ShouldBe("Message Foo.");
-            result.Locations[0].PhysicalLocation.Region.StartLine.ShouldBe(23);
-            result.BaselineState.ShouldBe(BaselineState.Updated);
-
-            result = run.Results[1];
-            result.Message.Text.ShouldBe("Message Foo.");
-            result.Locations[0].PhysicalLocation.Region.StartLine.ShouldBe(123);
-            result.BaselineState.ShouldBe(BaselineState.Updated);
-
-            result = run.Results[2];
-            result.Message.Text.ShouldBe("Message Foo Bar.");
-            result.Locations[0].PhysicalLocation.Region.StartLine.ShouldBe(1234);
+            var run = sarifLog.Runs.ShouldHaveSingleItem();
+            var result = run.Results.ShouldHaveSingleItem();
             result.BaselineState.ShouldBe(BaselineState.Updated);
         }
 
         [Fact]
-        public void Should_Generate_Report_With_BaseLineState_None_If_BaselineGuid_Not_Set()
+        public void Should_Generate_Report_With_BaseLineState_None_If_BaselineGuid_Is_Not_Set()
```

Comment:
Should we have two tests to check the logic?
- one with no guid set with existing issues (which should be ignored)
- one with a guid but not existing issues..